py/private/pytest.py.tmpl

```diff
@@ -52,7 +65,7 @@ if __name__ == "__main__":
     if len(cli_args) > 0:
         args.extend(cli_args)
 
-    exit_code = pytest.main(args)
+    exit_code = pytest.main(args, plugins=[ShardPlugin()])
```

Comment:
maybe we should avoid installing the plugin if sharding wasn't enabled (or the shard count is 1) ?
Helps to reduce the blast radius in case there's a problem with the plugin, it's easy for users to avoid going in that codepath